ui/apps/platform/cypress/integration/integrations/imageIntegrations.test.js

```diff
@@ -170,6 +170,63 @@ describe('Image Integrations', () => {
         cy.get('label:contains("Use container IAM role")').should('not.exist');
     });
 
+    it('should create a new Google Artifact Registry integration', () => {
+        const integrationName = generateNameWithDate('Google Artifact Registry Test');
+        const integrationType = 'artifactregistry';
+
+        visitIntegrationsTable(integrationSource, integrationType);
+        clickCreateNewIntegrationInTable(integrationSource, integrationType);
+
+        // Step 0, should start out with disabled Save and Test buttons
+        cy.get(selectors.buttons.test).should('be.disabled');
+        cy.get(selectors.buttons.save).should('be.disabled');
+
+        // Step 1, check empty fields
+        getInputByLabel('Integration name').type(' ');
+        getInputByLabel('Registry endpoint').type(' ');
+        getInputByLabel('Project').type(' ');
+        getInputByLabel('Service account key (JSON)').type(' ').blur();
+
+        getHelperElementByLabel('Integration name').contains('An integration name is required');
+        getHelperElementByLabel('Registry endpoint').contains('An endpoint is required');
+        getHelperElementByLabel('Project').contains('A project is required');
+        getHelperElementByLabel('Service account key (JSON)').contains(
+            'A service account key is required'
+        );
+        cy.get(selectors.buttons.test).should('be.disabled');
+        cy.get(selectors.buttons.save).should('be.disabled');
+
+        // Step 2, check conditional fields
+
+        // Step 2.1, enable workload identity, this should remove the service account field
+        getInputByLabel('Use workload identity').click();
+        cy.get(
+            `.pf-c-form__group:has('.pf-c-form__control:contains("Service account key (JSON)")') textarea`
+        ).should('not.exist');
+        // Step 2.2, disable workload identity, this should render the service account field again
+        getInputByLabel('Use workload identity').click();
+        getInputByLabel('Service account key (JSON)').should('be.visible');
```

Comment:
If this is counterpart to `'not.exist'` in line 205, then you might replace `'be.visible'` with `'exist'`

Aha, that reminds me to follow up on improvement to helper function so you can use it for both positive and negative assertions. I think we saw similar asymmetrical assertions recently in a contribution by Daniel.